src/root_finding/root_finding.jl

```diff
@@ -1,39 +1,127 @@
 
+# immutable Root{T<:Real}
```

Comment:
I haven't benchmarked it. The change was necessary to have more readable code (and results). I expect it doesn't change the speed too much (at least on v0.4).